File: app/src/main/java/resource/com/br/santanderapp/ui/investment/InvestmentInfoAdapter.java
package resource.com.br.santanderapp.ui.investment;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import resource.com.br.santanderapp.R;
import resource.com.br.santanderapp.model.investmentModel.Info;

public class InvestmentInfoAdapter extends RecyclerView.Adapter<InvestmentInfoAdapter.InvestmentViewHolder> {
    private Context context;
    private List<Info> listInfo;


    public InvestmentInfoAdapter(Context context, List<Info> listInfo) {
        this.context = context;
        this.listInfo = listInfo;
    }

    @NonNull
    @Override
    public InvestmentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View createView = LayoutInflater.from(context).inflate(R.layout.item_investment, parent, false);
        return new InvestmentViewHolder(createView);
    }

    @Override
    public void onBindViewHolder(@NonNull InvestmentViewHolder holder, int position) {
        Info info = listInfo.get(position);
        holder.settingsData(info);
    }

    @Override
    public int getItemCount() {
        return listInfo.size();
    }

    public class InvestmentViewHolder extends RecyclerView.ViewHolder {
        private Info listInfo;
        private TextView name;
        private TextView data;

        public InvestmentViewHolder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.item_investment_description);
            data = itemView.findViewById(R.id.item_investment_value);

        }

        public void settingsData(Info listInfo) {
            this.listInfo = listInfo;
            fillFields(listInfo);
        }

        private void fillFields(Info listInfo) {
            name.setText(listInfo.getName());
            data.setText(listInfo.getData());
        }
    }
}

File: app/src/main/java/com/rafhack/testeandroid/investment/InvestmentInfoAdapter.kt
package com.rafhack.testeandroid.investment

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.rafhack.testeandroid.R
import com.rafhack.testeandroid.data.entities.investment.Info

class InvestmentInfoAdapter(private val infoList: ArrayList<Info>) : RecyclerView.Adapter<InvestmentInfoAdapter.ViewHolder>() {

    companion object {
        const val typeInfo = 0
        const val typeDownload = 1
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(when (viewType) {
            typeInfo -> R.layout.investment_info_item
            typeDownload -> R.layout.investment_down_info_item
            else -> 0
        }, parent, false))
    }

    override fun getItemViewType(position: Int): Int {
        return if (infoList[position].data == null) typeDownload else typeInfo
    }

    override fun getItemCount(): Int = infoList.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(position)
    }

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        fun bind(position: Int) {

            val tvwName = itemView.findViewById<TextView>(when (itemViewType) {
                typeInfo -> R.id.investment_info_tvw_name
                typeDownload -> R.id.investment_down_info_tvw_name
                else -> 0
            })

            val tvwData = itemView.findViewById<TextView>(when (itemViewType) {
                typeInfo -> R.id.investment_info_tvw_data
                typeDownload -> R.id.investment_down_info_tvw_download
                else -> 0
            })

            val info = infoList[position]
            tvwName.text = info.name
            if (itemViewType == typeInfo) tvwData.text = info.data
        }
    }

}